safe/impact_function/impact_function.py

```diff
@@ -739,6 +740,24 @@ def _check_layer(layer, purpose):
             return PREPARE_FAILED_BAD_INPUT, message
 
         layer.keywords = keywords
+
+        if is_vector_layer(layer):
+            try:
+                check_inasafe_fields(layer, keywords_only=True)
+            except InvalidLayerError:
+                parameters = {
+                    'source': layer.publicSource()
+                }
+                message = generate_input_error_message(
+                    tr('The %s layer is not up to date.' % purpose),
```

Comment:
We should put `tr('foo bar hojah') % purpose` to make the translation works properly. It's better to use `format` also.